app/main.py

```diff
@@ -2,5 +2,16 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+
+    cached_results_storage = {}
+
+    def inner(*args, **kvargs) -> int:
+        if args in cached_results_storage:
+            print("Getting from cache")
+            return cached_results_storage.get(args)
+        else:
```

Comment:
Why exactly is it redundant? Since I want two different codes to execute depending on the result of a condition.

Reply:
The code that is written in the block "else" can be taken out of the conditional block and simply written below at the same level as the "if" statement. This will not affect the result but will make your code better!

Reply:
I agree that it's good code style to not use else, if not needed. but in my case it's needed. Because I don't want my program to print "Calculating new result" when 0 positional arguments are passes to function.

Reply:
If you take the code out of "else" block, it will still be executed only when the block of code in "if" has not been executed. The function will complete execution if it enters "if" due to "return".